### PR TITLE
Clairvoyance v2.0.0 - Package, Async, Linting

### PR DESCRIPTION
**Pull request by c3b5aw**
# Changes

## Repo structure
- feat: add linting suite (mypy, pylint, yapf)
- feat: added vscode configuration (hiding file, auto run macros .. )
- feat: default wordlist

## Docker
- refactor: migrated to a poetry version (68mb total size with the wordlist)

### cd.yml
- feat: add auto release both on Github and PyPi
     -> Require `${{ secrets.PYPI_TOKEN }}` to be set.
- chore: renamed cicd-to-dockerhub.yml -> cd.yml

### test.yml
- feat: add linting pipeline
- chore: updated invokation of pytest

## Code

- refactor: inverted quotes (That is pure feeling, we can revert it)
- feat: async client
- feat: async pooling of tasks (fields resolution, type resolution ...)
- feat: auto logger, ability to pass logger
- feat: client stored headers
- refactor: config as a class
- fix: reduced default bucket size from 4096 to 64
- feat: add field regex, corrected logical statement
- feat: add verbose argument
- chore: remove requests, add aiohttp and asyncio

# ToDo

- [x] chore: typo regexs -> regexes
- [x] feat: Verify options integrity
- [ ] feat: Client Task Consumer (Must create tasks from a generator and stream responses)
- [x] feat: Configuration of Async Client
- [x] feat: Accept env var to configure log_fmt, date_fmt
- [x] test: green tests

### CODE REVIEW

**Comment by c3b5aw** [comment]
@nikitastupin 
Please, highlight changes your disagree with.
Suggest the features you wants, I'll work on the todos.

File: docker-entrypoint.sh
Comment (nikitastupin) [inline docker-entrypoint.sh:7]: I wonder why do we need `exec` here 🤔

File: pyproject.toml
Comment (nikitastupin) [inline pyproject.toml:6]: Would you be willing to include your name too?

**Comment by nikitastupin** [comment]
Hey @c3b5aw,

Thank you very much for the PR. I've skimmed the changes and left few comments. Other than that we're good.

Regarding further feature requests, I suggest to merge this one as soon as possible (after finishing the ToDo of course) and then add things iteratively (it's much easier and faster to review small PRs). Would you be willing to proceed this way?

I can go through existing Issues and Pull Requests, de-duplicate and close stale ones so that we'll know what to focus on next.

File: pyproject.toml
Comment (nikitastupin) [reply pyproject.toml:6]: Or is there the `contributors` field?

File: .github/workflows/cd.yml
Comment (nikitastupin) [inline .github/workflows/cd.yml:41]: We don't need this condition with `push`

File: pyproject.toml
Comment (iCarossio) [reply pyproject.toml:6]: Yeah I defintely think you deserve it @c3b5aw ;)

Comment (iCarossio) [reply pyproject.toml:6]: > Or is there the `contributors` field?

https://python-poetry.org/docs/pyproject/
No there is only "authors" and "maintainers". 

Concerning the "author" field, please know that we started working on improving Clairvoyance about 8 months ago with @MdotTIM (we all work for https://escape.tech). However we never reached a sufficient state of quality to give you back those changes ;)
@c3b5aw restarted completely from your repo and reached this amazing results during his Open Source Days :) I absolutely think he deserves being co-author with you on this v2.

At the moment you are officially the only maintainer of the repo but we love contributing to the GraphQL Ecosystem so we could help maintaining the repo if this is something you're okay with ;) 

Best,

Antoine
Co-founder & CTO @Escape.tech

File: docker-entrypoint.sh
Comment (c3b5aw) [reply docker-entrypoint.sh:7]: It's a docker practice allowing to forward everything (signals, env, ...) to the new shell.

You can read more about this:
https://stackoverflow.com/questions/39082768/what-does-set-e-and-exec-do-for-docker-entrypoint-scripts

File: .github/workflows/cd.yml
Comment (c3b5aw) [reply .github/workflows/cd.yml:41]: This is on `push tags`.
I assume that when you push a new tag to github, it release everything (and follow the tag versioning).

**Comment by c3b5aw** [comment]
> Hey @c3b5aw,
> 
> Thank you very much for the PR. I've skimmed the changes and left few comments. Other than that we're good.
> 
> Regarding further feature requests, I suggest to merge this one as soon as possible (after finishing the ToDo of course) and then add things iteratively (it's much easier and faster to review small PRs). Would you be willing to proceed this way?
> 
> I can go through existing Issues and Pull Requests, de-duplicate and close stale ones so that we'll know what to focus on next.

Agreed

File: .github/workflows/cd.yml
Comment (nikitastupin) [reply .github/workflows/cd.yml:41]: Yep, that's why I suggest to remove the `if` condition because it doesn't make sense in the context of the `push` event.

Comment (nikitastupin) [reply .github/workflows/cd.yml:41]:
```suggestion
```

File: .github/workflows/cd.yml
Comment (nikitastupin) [inline .github/workflows/cd.yml:12]:
```suggestion
```

The `uses` key is pretty self-explanatory in this case

**Comment by nikitastupin** [comment]
Thank you! I'm going to review and merge closer to the end of this week.

**Review by nikitastupin — APPROVED** [review]
I'll make some minor changes after the update but overall looks good!

**Comment by nikitastupin** [comment]
Seems to work fine with exception of publishing to PyPI - the "clairvoyance" name is already claimed. I've filled the details at https://github.com/nikitastupin/clairvoyance/issues/42 and probably start solving it next week.

**Pull request merged**
